### PR TITLE
Remove storage handles

### PR DESCRIPTION
**Pull request by jblebrun**
Had to move RawEntityDereferencerTest to the proper place.

All functionality in tests deleted here should be covered elsewhere.

### CODE REVIEW

File: java/arcs/android/storage/handle/AndroidHandleManager.kt [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-import kotlin.coroutines.EmptyCoroutineContext
-
-/**
- * AndroidHandleManager will create a [HandleManager] instance, replacing the default
- * [ActivationFactory] with one that generates [ServiceStore] instances that can
- * communication with a running [StorageService].
- */
-@Deprecated("Use arcs.core.host.EntityHandleManager")
-@OptIn(kotlinx.coroutines.ExperimentalCoroutinesApi::class)
-fun AndroidHandleManager(
```
Comment (galganif) [inline java/arcs/android/storage/handle/AndroidHandleManager.kt]: does this mean you cannot create handles in Android without using an archost?

Comment (jblebrun) [reply java/arcs/android/storage/handle/AndroidHandleManager.kt]: You can, just use
```
val androidHandleManager = EntityHandleManager(
  time = JvmTime,
  activationFactory = ServiceStoreFactory(context, lifecycle)
)
```

**Comment by jblebrun** [comment]
copybara fix: cl/303643209

**Comment by jblebrun** [comment]
I forgot to clean up the test directories :( cl/303654755

**Pull request merged**